app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java

```diff
@@ -2079,4 +2079,31 @@ public void updateMysqlPostgresMongoSslMode(MongoTemplate mongoTemplate) {
                     }
                 });
     }
+
+    @ChangeSet(order = "062", id = "create-entry-in-sequence-per-organization-for-datasource", author = "")
+    public void createEntryInSequencePerOrganizationForDatasource(MongoTemplate mongoTemplate) {
+
+        Map<String, Long> maxDatasourceCount = new HashMap<>();
+        mongoTemplate
+                .find(query(where("name").regex("^Untitled Datasource \\d+$")), Datasource.class)
+                .forEach(datasource -> {
+                    long count = 1;
+                    String datasourceCnt = datasource.getName().replaceAll("[^\\d]", "").trim();
```

Comment:
How about just `.substring` on it, instead of invoking regex?

```suggestion
                    String datasourceCnt = datasource.getName().substring("Untitled Datasource ".length);
```

Reply:
Should I assume that there won't be any extra spaces after the sequenceCount? So that I won't have to use trim() as you suggested. 
e.g. "Untitled Datasource {count}"

Reply:
The regex we are using in the query has a `$` after `\d+` right? So it'll find only those that end with the number and there's nothing after the number. So yeah, that's a fair assumption to make I think.